bigquery_etl/shredder/delete.py

```diff
@@ -249,6 +249,18 @@ def create_job(client):
                 f"_source_{index} IS NULL" for index, _ in enumerate(sources)
             )
 
+            if partition.id is None:
+                condition = f"""
+                (
+                  ({partition.condition}) IS NOT TRUE
+                  OR ({field_conditions})
+                )
+                """
+                if partition.required_partition_filter:
+                    condition += f" AND ({partition.required_partition_filter})"
+            else:
+                condition = f"({field_conditions}) AND ({partition.condition})"
```

Comment:
The `partition.required_partition_filter` stuff seems a bit kludgy.  If the problem is that the `({field_conditions})` branch of the `OR` doesn't have a partition condition then I think we could simply include `AND ({partition.condition})` there, and potentially simplify this into something like:
```suggestion
            condition = f"({field_conditions}) AND ({partition.condition})"
            if partition.id is None:
                condition = f"({condition}) OR (({partition.condition}) IS NOT TRUE)"
```
(the parentheses around `({partition.condition}) IS NOT TRUE` wouldn't technically be required, but I think it'd help avoid potential confusion about whether the `IS NOT TRUE` applied to the result of the `OR` expression)

Reply:
I just checked, and this doesn't satisfy `require_partition_filter`

Reply:
how about this solution that doesn't use `required_partition_filter`, and satisfies `require_partition_filter`:
```suggestion
            if partition.id is None:
                # only apply field conditions on partition.condition
                field_conditions = f"""
                ({partition.condition}) IS NOT TRUE
                OR ({field_conditions})
                """
                # always true partition condition to satisfy require_partition_filter
                partition_condition = f"""
                ({partition.condition}) IS NOT TRUE
                OR ({partition.condition})"
                """
            else:
                partition_condition = partition.condition
                
            condition = f"({field_conditions}) AND ({partition_condition})"
```